### PR TITLE
Better UDF support

### PR DESCRIPTION
**Pull request by Jolanrensen**
from issue: https://github.com/JetBrains/kotlin-spark-api/issues/143

One gotcha I noticed is that
```kotlin
val toNormalClass2 by udf.register { a: String, b: Int ->
    NormalClass(b, a)
}
shouldThrow<AnalysisException> { // toNormalClass2 is never accessed, so the delegate getValue function is not executed
    spark.sql("select toNormalClass2(first, second) from test2").show()
}
```
won't work since `toNormalClass2` is never accessed, so the register function is not executed. A simple `val a = toNormalClass2` already fixes this, but it's not optimal...

### CODE REVIEW

**Comment by Jolanrensen** [comment]
Maybe a vararg typed udf as final one?

**Comment by Jolanrensen** [comment]
updated wiki https://github.com/Kotlin/kotlin-spark-api/wiki/UDF

**Comment by Jolanrensen** [comment]
@asm0dey whenever you have time to check it :)

File: kotlin-spark-api/3.2/src/main/kotlin/org/jetbrains/kotlinx/spark/api/UDFDelegates.kt [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+//    UDFWrapperBuilderDelegate { name ->
+//        register(
+//            name = name,
+//            asNondeterministic = asNondeterministic,
+//            func = func,
+//        )
+//    }
+//
+///** Creates delegate of [UDFRegistration]. Use like:
+// * ```kotlin
```
Comment (asm0dey) [inline kotlin-spark-api/3.2/src/main/kotlin/org/jetbrains/kotlinx/spark/api/UDFDelegates.kt:357]: Looks like this file should be deleted

**Comment by asm0dey** [comment]
It's definitely 1.2.0, not 1.1.1

**Comment by Jolanrensen** [comment]
For future reference, these files were used to generate the UDFs:

[generateUDF.zip](https://github.com/Kotlin/kotlin-spark-api/files/9001870/generateUDF.zip)

**Pull request merged**